### ISSUE TITLE
Sqp

### ISSUE THREAD

**Issue by ifuchs** [pyodide/pyodide#2233]
I’m looking for a way to deploy a webapp that can access a local sql server (on same network as the client). Can pyodide be used to do this?

**Comment by ryanking13** [comment]
One blocker for that usage is that Pyodide [doesn't support synchronous I/O](https://pyodide.org/en/stable/project/roadmap.html#synchronous-io) for now. But I think you can try asynchronous libraries such as aiomysql to mitigate it.

**Comment by hoodmane** [comment]
Or you can use my experimental but generally working library [synclink](github.com/hoodmane/synclink) if you are adventurous.

**Comment by ifuchs** [comment]
I am new to pyodide. Assuming it is possible, can someone please say more about how exactly I can communicate with an ms sql server from pyodide?

**Comment by ryanking13** [comment]
> I am new to pyodide. Assuming it is possible, can someone please say more about how exactly I can communicate with an ms sql server from pyodide?

Hmm, actually no. I tried to connect to sql server directly, but it doesn't seem possible. The main blocker is that it requires socket connection when connecting to SQL server, which is not possible in Pyodide.

**Comment by ifuchs** [comment]
Ok, thanks for trying.

**Comment by ryanking13** [comment]
@Pancham97

No, accessing database requires socket connection and it is not supported in Pyodide.

**Comment by Pancham97** [comment]
> @Pancham97
> 
> 
> 
> No, accessing database requires socket connection and it is not supported in Pyodide.

Okay, would you recommend any other way to do so @ryanking13?

**Comment by ryanking13** [comment]
Well, why do you want to access the database in the browser? It will leak everything, including passwords and queries.

**Comment by Pancham97** [comment]
> Well, why do you want to access the database in the browser? It will leak everything, including passwords and queries.

I think that as a Python user, I should have the full capacity of using Python packages that talk to databases in the notebook environment. Whether or not I would like to use it for production environment is a different thing. Perhaps, users may use it to demonstrate a quick demo of SQL queries? WDYT?

**Comment by ryanking13** [comment]
I understand that there may be uses for it, but it's still technically challenging.

I think what you can do for now is one of the following:

1. make a middleware that works between the client (browser, pyodide) and DB that handles HTTP requests in the middle.

2. find a DBMS that can work asynchronously with a simple HTTP request/response (I don't know if such a thing exists).

3. or use an in-memory database that doesn't require network communication, such as sqlite3..
